tests/moduletool/test_update.py

```diff
@@ -183,6 +183,64 @@ def assert_version_installed(module_name: str, version: str) -> None:
     assert ModuleV1(project=None, path=mod11_dir).version == Version("4.1.2")
 
 
+@pytest.mark.slowtest
+def test_module_update_dependencies(
+    tmpdir: py.path.local,
+    snippetcompiler_clean,
+    modules_dir: str,
+) -> None:
+    """
+    Verify that `inmanta project update` correctly handles module's Python dependencies:
+        - update should include install
+        - update should update Python dependencies within module's constraints
+        - update should update transitive Python dependencies
+    """
+    snippetcompiler_clean.setup_for_snippet(
+        snippet="import my_mod",
+        autostd=False,
+        install_project=False,
+        add_to_module_path=[str(tmpdir.join("modules"))],
+    )
+
+    # create index with multiple versions for packages a, b and c
+    index: PipIndex = PipIndex(str(tmpdir.join("index")))
+    create_python_package("a", Version("1.0.0"), str(tmpdir.join("a-1.0.0")), publish_index=index)
+    for v in ("1.0.0", "1.0.1", "2.0.0"):
+        create_python_package(
+            "b", Version(v), str(tmpdir.join(f"b-{v}")), requirements=[Requirement.parse("c")], publish_index=index
+        )
+    for v in ("1.0.0", "2.0.0"):
+        create_python_package("c", Version(v), str(tmpdir.join(f"c-{v}")), publish_index=index)
+
+    # install b-1.0.0 and c-1.0.0
+    process_env.install_from_index([Requirement.parse(req) for req in ("b==1.0.0", "c==1.0.0")], index_urls=[index.url])
+
+    # create my_mod
+    v1_module_from_template(
+        source_dir=os.path.join(modules_dir, "minimalv1module"),
+        dest_dir=str(tmpdir.join("modules", "my_mod")),
+        new_name="my_mod",
+        new_requirements=[Requirement.parse(req) for req in ("a", "b~=1.0.0")],
+    )
+
+    # run `inmanta project update` without running install first
+    environ_index: Optional[str] = os.environ.get("PIP_INDEX_URL", None)
```

Comment:
The monkeypatching fixture has faciltities to update env vars as well.

Reply:
Interesting, I thought I couldn't use that because it's outside of the Python scope. Very nice, will fix.